src/core/init/stream_events_emitter/types.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface IStreamEventData {
+  start: number;
+  end?: number;
+  id?: string;
+  isBeingPlayed: boolean;
```

Comment:
That sounds like a solution where a WeakMap may be a better solution.

Reply:
Tried to do it. Done